gateway-js/src/index.ts

```diff
@@ -563,28 +562,25 @@ export class ApolloGateway implements GraphQLService {
       this.logger.info('Updated Supergraph SDL was found.');
     }
 
-    await this.maybePerformServiceHealthCheck(result);
-
-    this.compositionId = result.id;
-    this.supergraphSdl = result.supergraphSdl;
+    this.compositionId = id;
+    this.supergraphSdl = supergraphSdl;
     this.parsedSupergraphSdl = parsedSupergraphSdl;
 
-    const { schema, supergraphSdl } = this.createSchemaFromSupergraphSdl(
-      result.supergraphSdl,
-    );
+    const { schema, supergraphSdl: generatedSupergraphSdl } =
+      this.createSchemaFromSupergraphSdl(supergraphSdl);
 
-    if (!supergraphSdl) {
+    if (!generatedSupergraphSdl) {
       this.logger.error(
         "A valid schema couldn't be composed. Falling back to previous schema.",
       );
     } else {
-      this.updateWithSchemaAndNotify(schema, supergraphSdl);
+      this.updateWithSchemaAndNotify(schema, generatedSupergraphSdl);
 
       if (this.experimental_didUpdateComposition) {
```

Comment:
Should deprecate this (possibly remove it) in favor of something like a `didUpdateSupergraph` since composition is no longer the gateway's concern